dpdata/unit.py

```diff
@@ -0,0 +1,169 @@
+from abc import ABC
+from scipy import constants
+
+AVOGADRO = constants.Avogadro                             # Avagadro constant
+ELE_CHG  = constants.elementary_charge                    # Elementary Charge, in C
+BOHR     = constants.value("atomic unit of length")       # Bohr, in m
+HARTREE  = constants.value("atomic unit of energy")       # Hartree, in Jole
+RYDBERG  = constants.Rydberg * constants.h * constants.c  # Rydberg, in Jole
+
+# energy conversions
+econvs = {
+    "eV": 1.0,
+    "hartree": HARTREE / ELE_CHG,
+    "kJ_mol": 1 / (ELE_CHG * AVOGADRO / 1000),
+    "kcal_mol": 1 / (ELE_CHG * AVOGADRO / 1000 / 4.184),
+    "rydberg": RYDBERG / ELE_CHG,
+    "J": 1 / ELE_CHG,
+    "kJ": 1000 / ELE_CHG
+}
+
+# length conversions
+lconvs = {
+    "angstrom": 1.0,
+    "bohr": BOHR * 1E10,
+    "nm": 10.0,
+    "m": 1E10,
+}
+
+def check_unit(unit):
+    if unit not in econvs.keys() and unit not in lconvs.keys():
+        try:
+            eunit = unit.split("/")[0]
+            lunit = unit.split("/")[1]
+            if eunit not in econvs.keys():
+                raise RuntimeError(f"Invaild unit: {unit}")
+            if lunit not in lconvs.keys():
+                raise RuntimeError(f"Invalid unit: {unit}")
+        except:
+            raise RuntimeError(f"Invalid unit: {unit}")
+
+class Conversion(ABC):
+    def __init__(self, unitA, unitB, check=True):
+        """
+        Parent class for unit conversion
+
+        Parameters
+        ----------
+            unitA : str, unit to be converted
+            unitB : str, unit which unitA is converted to, i.e. `1 unitA = self._value unitB`
+            check : bool, whether to check unit validity
+        
+        Examples
+        --------
+        >>> conv = Conversion("foo", "bar", check=False)
+        >>> conv.set_value("10.0")
+        >>> print(conv)
+        >>> "1 foo = 10.0 bar"
```

Comment:
output value should be in this format
```suggestion
        1 foo = 10.0 bar
```